woocommerce/woocommerce-ios
# [Woo POS] Update totals view UI with subtotal from cart total

Part of #12763 


## Why

To make the integration with data layer (WIP) easier, this PR updates the totals view UI so that we have the views ready to show the order **subtotal**, **total**, and **tax** (highlighted in the screenshot below). I'm making the assumption that the "subtotal" field means the cart total, please feel free to correct it if you have a different understanding. The **total** and **tax** fields will come from the order synced remotely in a future PR. I didn't spend much time styling the UI to focus on the end-to-end flow first.

<img width="463" alt="Screenshot 2024-05-28 at 3 12 27 PM" src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/d070fd74-34ce-47ed-8ce9-84f8e3080633">

## How

First, the diff in`CartView`'s id change in https://github.com/woocommerce/woocommerce-ios/pull/12850/commits/2571a392e7dd917e2500eb44d6be1d8bce75c749 is to fix a SwiftUI warning on duplicated ID when adding more than one of the same product.

The main UI updates are in `TotalsView`, replacing the cart item list with the 3 price fields with very rough styling.

To add a bit more value to the PR, a new observable `formattedCartTotalPrice` property was added to `PointOfSaleDashboardViewModel` to show the "subtotal" field in the totals view. It observes the products in the cart and sums up the total using a DI'ed currency formatter. This is unit-testable but I'm leaving it for a later subtask when more product decisions are finalized/clarified like what this "subtotal" means and whether we're supporting quantity update in the cart.

## Testing instructions


Prerequisite: having a store eligible for POS

- Switch to a store in the prerequisite if needed
- Turn on the feature switch in Menu > Settings > Experimental Features > POS, if needed
- Go to the Menu tab --> the POS row should appear shortly
- Tap to add some products to the cart
- Tap `Checkout` --> the totals view should show the cart total in the "Subtotal" price field
- Feel free to go back to update the cart and then check the "Subtotal" price field after tapping `Checkout` again

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/1945542/b3bb7eb6-4dbd-45f9-b7e7-ee092dc3c0ca


<img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/ac231b1b-690b-4a8a-a267-8e64a497978e" width="400" />

---
